library/x509_crt.c

```diff
@@ -1575,9 +1589,13 @@ int mbedtls_x509_crt_parse_path(mbedtls_x509_crt *chain, const char *path)
             continue;
         }
 
+        if (FAILED(SizeTToInt(wcslen(file_data.cFileName), &length_as_int))) {
+            return MBEDTLS_ERR_X509_FILE_IO_ERROR;
+        }
+
         w_ret = WideCharToMultiByte(CP_ACP, 0, file_data.cFileName,
-                                    -1,
-                                    p, (int) len,
+                                    length_as_int,
+                                    p, (int) len - 1,
```

Comment:
(sorry, just noticed) below at (new) line 1620 we have "mbedtls_x509_crt_parse_path not available before Windows XP". Since we no longer support <= WINXP, can we delete these checks?